packages/@aws-cdk/aws-route53-targets/lib/cloudfront-target.ts

```diff
@@ -11,12 +12,36 @@ export class CloudFrontTarget implements route53.IAliasRecordTarget {
    */
   public static readonly CLOUDFRONT_ZONE_ID = 'Z2FDTNDATAQYW2';
 
+  /**
+   * Get the hosted zone id for the current scope.
+   *
+   * @param stack - Stack in which the resource is defined
+   */
+  public static getHostedZoneId(stack: Stack) {
+    if (!CloudFrontTarget.stackPartitionHostedZoneMapping.hasOwnProperty(stack.stackId)) {
+      CloudFrontTarget.stackPartitionHostedZoneMapping[stack.stackId] = new CfnMapping(stack, 'AWSCloudFrontPartitionHostedZoneIdMap', {
```

Comment:
is this mapping guaranteed to be created once per stack? or is it possible to have duplicates if there are multiple cloudfront targets for example..

Reply:
This should be covered by the `stackPartitionHostedZoneMapping` Object and the corresponding assignment within the `getHostedZoneId` method using the `stackId`. The proper handling of nested stacks is also covered by the `use CloudFrontTarget hosted zone id mappings in nested stacks` [test case](https://github.com/aws/aws-cdk/pull/9174/files#diff-29ef1546ef1fe5d915a766cc16adbdfbR31-R56).